src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -517,6 +530,12 @@ public UInt256 ReserveOwnTransactionNonce(Address address)
             return currentNonce;
         }
 
+        public UInt256? GetLatestPendingNonce(Address address)
+        {
+            _nonces.TryGetValue(address!, out AddressNonces addressNonces);
```

Comment:
Wouldn't it be better to just grab latest nonce directly from _transactions for a specified address?